File: Ass2Q3.py
def store(am):
    Q=[]
    L=[]
    with open(am,'r+') as file:
        i=0
        while i<3:
            s=file.readline()
            p=s.split()
            j=0
            while j<3:
                L.append(float(p[j]))
                j=j+1
            Q.append(L)
            L=[]
            i=i+1

        return Q
M=store("a.txt")
N=store("b.txt")
p=[[],[],[]]
t=0
for i in range(3):
    for j in range(3):
        for k in range(3):
            t+=M[i][k]*N[k][j]
        p[i].append(t)
        t=0
    j=0
print('\n')
print("The product of the two Matrices is MxN: ")
for i in range(3):
    if(i>0):
        print('\n')
    for j in range(3):
        print(p[i][j],end="   ")
print('\n')
A=[-1,5,8]
p2=[]
s=0
for a in range(3):
    for b in range(3):
        s+= M[a][b]*A[b]
    p2.append(s)
    s=0
print("The product of M and vector {0} is :\n{1}".format(A,p2))
'''The product of the two Matrices is MxN:
1.0   1.0   2.0

1.0   1.0   2.0

0.0   2.0   2.0

The product of M and vector [-1, 5, 8] is :
[4.0, 7.0, 13.0]'''

File: Ass2Q3.cpp
#include<iostream>
#include<fstream>
#include<string>
#include<vector>
using namespace std;
vector<vector<float>> store(string am){
    vector<vector<float>> Q;
    vector<float> L;
    float K[3];
    ifstream file(am);
    if (file.is_open()){
       for(int i=0;i<3;i++){ 
            for(int i=0;i<3;i++){
                file>>K[i];
            }
            for(int i=0;i<3;i++){
                L.push_back(K[i]);
            }
            Q.push_back(L);
            L={};
       }
    }
    return Q;

}
int main()
{  vector<vector<float>> M=store("a.txt");//This is where you enter the name of the text file
    vector<vector<float>> N=store("b.txt");
  vector<vector<float>> p={{},{},{}};
    float t=0;
    for (int i=0;i<3;i++){
     for (int j=0;j<3;j++){
      for (int k=0;k<3;k++){
              t+=M[i][k]*N[k][j];
            }
            p[i].push_back(t);
            t=0;
        }  
    }
    cout<<endl;
       cout<<"The product of the two Matrices is MxN: ";
    for(int i=0; i<3;i++){
        if(i>0) {
            cout<<endl;
            cout<<"                                        ";
        }
        for(int j=0; j<3;j++){
            cout<<p[i][j]<<"   ";
        }
    }
    vector<float> A={-1,5,8};
    vector<float> p2;
    float s=0;
    for (int a=0;a<3;a++){
        for (int b=0;b<3;b++){
            s+= M[a][b]*A[b];
        }
        p2.push_back(s);
        s=0;
    }
    
    cout<<"\nThe product of M and vector A={-1,5,8} is:";
    for (int i=0;i<4;i++){
        if (i==0) cout<<"(";
        else {
            if (i!=3) cout<<p2[i-1]<<",";
            else cout<<p2[i-1]<<")";
        }
    }
cout<<'\n';
return 0;}
/* M=1 1 0   N=1 0 1
     1 0 1     0 1 1
     0 1 1     0 1 1  
MXN=1   1   2   MXA=(4,7,13)
    1   1   2   
    0   2   2  */
   
   
